File: Java/simple/ForNullPointerBug/src/at/faultycode/java/simple/ForNullPointerBug.java
package at.faultycode.java.simple;

import java.util.logging.Logger;

/**
 * @author MaAb
 */
public class ForNullPointerBug {

	/** Default Java logger **/
	private static final Logger LOGGER = Logger
			.getLogger(ForNullPointerBug.class.getName());

	private ForNullPointerBug() {}
	
	/**
	 * Entry point of program
	 * 
	 * @param args
	 */
	public static void main(String[] args) {
		String message = null;
		for (int i = 0; i < args.length; i++) {
			message += String.format("%d: %s; ", i, args[i]);
		}

		if (!message.isEmpty()) {
			LOGGER.info(message);
		}
	}
}

File: C++/simple/ForNullPointerBug/src/ForNullPointerBug.cpp
///
/// Author: MaAb
///

namespace faulty {

struct A {
   int num;
};

int mayFailOrNot(int argc) {
    A *a = nullptr;
    for (int i = 1; i < argc; i++) {
        return 0;
    }
    return a->num;
}
}

///
/// Entry point of program
///
int main(int argc, char** argv) {
  return faulty::mayFailOrNot(argc);
}
